scripts/lib/CIME/hist_utils.py

```diff
@@ -421,21 +421,24 @@ def get_extension(model, filepath):
     """
     basename = os.path.basename(filepath)
     m = None
-    if model == "mom":
-        for ext in ('frc', 'sfc.day', 'prog', 'hmz', 'hm'):
-            regex_str = r'.*' + model + r'[^_]*_?([0-9]{4})?[.](' + ext + r'.?)([.].*[^.])?[.]nc'
-            ext_regex = re.compile(regex_str)
-            m = ext_regex.match(basename)
-            if m is not None:
-                break
-    elif model == 'cice':
-        ext_regex = re.compile(r'.*%s[^_]*_?([0-9]{4})?[.](h_inst.?)([.].*[^.])?[.]nc' % model)
-        m = ext_regex.match(basename)
-
-    if m is None:
-        ext_regex = re.compile(r'.*%s[^_]*_?([0-9]{4})?[.](h.?)([.].*[^.])?[.]nc' % model)
-        m = ext_regex.match(basename)
+    ext_regexes = []
 
+    # First add any model-specific extension regexes; these will be checked before the
+    # general regex
+    if model == "mom":
+        # Need to check 'sfc.day' specially: the embedded '.' messes up the
+        # general-purpose regex
+        ext_regexes.append(r'sfc\.day')
+
+    # Now add the general-purpose extension regex
+    ext_regexes.append(r'\w+')
+
+    for ext_regex in ext_regexes:
+        full_regex_str = model+r'\d?_?(\d{4})?\.('+ext_regex+r')[-\w\.]*\.nc\.?'
```

Comment:
This regex is the same as what was used prior to #2811 except that I am now using the variable ext_regex in place of '\w+', so that we can match different extensions with this same regex.